provisioning/device/src/Transports/Amqp/ProvisioningTransportHandlerAmqp.cs

```diff
@@ -65,6 +64,8 @@ internal ProvisioningTransportHandlerAmqp(ProvisioningClientOptions options)
 
             Argument.AssertNotNull(message, nameof(message));
 
+            using var connectionLostCancellationToken = new CancellationTokenSource();
```

Comment:
Making the connection close callback local, we can avoid making this a class member.

Reply:
I like this, so I'm making this same change in MQTT in #2797 FYI

Reply:
thank you for using it elsewhere :)